DataFormats/SoATemplate/interface/SoALayout.h

```diff
@@ -525,6 +525,7 @@
     /* data members */                                                                                                 \
     std::byte* mem_;                                                                                                   \
     size_type nElements_;                                                                                              \
+    size_type const scalar_ = 1;                                                                                       \
```

Comment:
Is the `scalar_` needed only for the iorule? Could it be `constexpr` or `static`?

Reply:
> Is the `scalar_` needed only for the iorule?

Yes.

> Could it be `constexpr` or `static`?

No, that doesn't work :-/